### PR TITLE
submit, bug fixes, dashboard messages

### PR DESCRIPTION
**Pull request by panjenny0**
### Dashboard message
- updated RSVP-by date to Jan 3rd 2021 (FYI @JPGarCar)
- updated copy text wording and embedded links
- fetching dates from firebase in `InternalWebsites/Livesite`: `sendAcceptancesBy`, `rsvpBy`, `offWaitlistNotify`

### [bugfix] Redirect to `/application/part-1` when application is not submitted
- added condition for `/application` in the `useEffect` of `Auth.js` to check if applicant has submitted their application before being able to access the dashboard - if their application is not submitted, redirect to `/application/part-1` (FYI @andykwan1124)
```
if (location === '/application') {
        await handleUser(setUser, setLocation)
      }
```
- I noticed that the user gets automatically redirected to `/submission` if they RSVP and refresh the page (or navigate away and come back) - I've restricted the redirect to `/submission` for HackCamp only

### [bugfix] User document fields get deleted when you click 'Back' several times in `/application`
- added `!applicationRef.current` in the first check in `forceSave` (thank you @jackyzha0 so big brained 🧠)
```
const forceSave = useCallback(async () => {
    if (!user || !applicationRef.current) return
    const updatedApp = {
      ...applicationRef.current,
      submission: {
        lastUpdated: firebase.firestore.Timestamp.now(),
        submitted: false,
      },
    }
```

### Other minor changes
- changed review page to display `otherEngagementSource` if `Other` is chosen for `engagementSource`
- changed width and border radius of review container to align with design specs
- also fixes bugs listed here https://github.com/nwplus/livesite/pull/220#issuecomment-743953305

#### testing dashboard redirect
1. go to https://nwplus-ubc--pr220-date-fix-yjvu589n.web.app/application/part-1 and login
2. go to `/application` - you should be redirected to part 1 of the app
note that after you click on 'submit' on the application, you should no longer get redirected

#### testing dashboard messages 
1. assuming you've logged in, access your Applicant document in the database: https://console.firebase.google.com/u/0/project/nwplus-ubc-dev/firestore/data~2FHackathons~2FnwHacks2021~2FApplicants
2. change your `applicantStatus` to `waitlisted` then go back to `/application` in the preview link https://nwplus-ubc--pr220-date-fix-yjvu589n.web.app/application and refresh - the status and message should update to waitlisted
3. change your `applicantStatus` to `accepted` then go back to preview link and refresh - message should update

### CODE REVIEW

**Review by kevin-zou — APPROVED** [review]
lgtm! 💯

**Review by vincentchiang1 — DISMISSED** [review]
💯

File: src/components/ApplicationForm/ReviewCards.js [hunk 1]
```diff
@@ -283,8 +294,26 @@ export default ({ formInputs, handleEdit, onChange }) => {
...
               })
             }
-            label="I authorize you to share my application/registration information for event administration, ranking, MLH administration, pre- and post-event informational e-mails, and occasional messages about hackathons in-line with the MLH Privacy Policy. I further agree to the terms of both the MLH Contest Terms and Conditions and the MLH Privacy Policy."
-          />
+          >
+            <span>
+              I authorize you to share my application/registration information for event
+              administration, ranking, MLH administration, pre- and post-event informational
+              e-mails, and occasional messages about hackathons in-line with the{' '}
+              <A bolded color="primary" href="https://mlh.io/privacy" target="_blank">
```
Comment (anneguo3) [inline src/components/ApplicationForm/ReviewCards.js:302]: omg just wanted to say than u for adding the links ❤️

**Comment by panjenny0** [comment]
Addressed some issues/bugs brought up by @anneguo3 (thank u!)
- [x] upon submission, go back to main page of portal (should not be able to see the pages anymore)
Screen Shot 2020-12-12 at 9 21 22 PM

- [x] Back on first page —> go “back to portal”
Screen Shot 2020-12-12 at 8 59 00 PM
_(I made this 'Save & Logout')_

- [x] limit infinite scroll on last page 
Screen Shot 2020-12-12 at 9 03 28 PM
_(lol this is the best I could do)_
- [x] add stars to the check boxes
- [x] default grad, number of hackathons as empty, and isOfLegalAge as null (unselected) 
Screen Shot 2020-12-12 at 8 20 43 PM
Screen Shot 2020-12-12 at 8 20 16 PM
_(to test this, delete your document in the db; the new changes should initialize null as initial values)_

File: src/utility/HackerApplicationContext.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function HackerApplicationProvider({ children }) {
   /**Saves the users application, can be called manually or through interval */
   /**Uses a reference to the application because I don't want all my useEffects triggering every time someone changes the application. */
   const forceSave = useCallback(async () => {
-    if (!user) return
+    if (!user || !applicationRef.current) return
```
Comment (jackyzha0) [inline src/utility/HackerApplicationContext.js:33]: bless up

File: src/utility/Constants.js [hunk 3]
```diff
@@ -101,6 +101,7 @@ export const hackerApplicationTemplate = Object.freeze({
 // relevant dates for each status message displayed on the dashboard
 export const relevantDates = Object.freeze({
   sendAcceptancesBy: 'December 30th, 2020',
-  rsvpBy: 'Janurary 7th at 11:59 PM (Pacific Time)',
+  rsvpBy: 'Janurary 3rd at 11:59 PM (Pacific Time)',
+  offWaitlist: 'one week before the event',
   hackathonDate: 'January 9-10th',
 })
```
Comment (jackyzha0) [inline src/utility/Constants.js:107]: is there a reason these aren't on firebase? livesite has some fields related to these

**Review by jackyzha0 — APPROVED** [review]

File: src/utility/Constants.js (hunk 3 above)
Comment (panjenny0) [reply src/utility/Constants.js:107]: Oh I totally didn't realize that there were similar fields on firebase
We were just a little short on time and I needed to make this change really quickly 😢 but let me just manually add some new fields and reference from firebase instead

Comment (panjenny0) [reply src/utility/Constants.js]: Ok updated! Pulling from firebase now

**Review by andykwan1124 — APPROVED** [review]
LGTM! nice fixes!!!!!

**Pull request merged**